syncopy/nwanalysis/connectivity_analysis.py

```diff
@@ -324,39 +326,28 @@ def connectivityanalysis(data, method="coh", keeptrials=False, output="abs",
     # Perform the trial-parallelized computation of the matrix quantity
     st_compRoutine.initialize(data,
                               st_out._stackingDim,
-                              chan_per_worker=None, # no parallelisation over channels possible
-                              keeptrials=keeptrials) # we most likely need trial averaging!
+                              chan_per_worker=None,   # no parallelisation over channels possible
+                              keeptrials=keeptrials)  # we most likely need trial averaging!
     st_compRoutine.compute(data, st_out, parallel=kwargs.get("parallel"), log_dict=log_dict)
 
-    # if ever needed..
     # for single trial cross-corr results <-> keeptrials is True
-    if keeptrials and av_compRoutine is None:
-        if out is not None:
-            msg = "Single trial processing does not support `out` argument but directly returns the results"
-            SPYWarning(msg)
+    if av_compRoutine is None:
         return st_out
 
     # ----------------------------------------------------------------------------------
     # Sanitize output and call the chosen ComputationalRoutine on the averaged ST output
     # ----------------------------------------------------------------------------------
 
-    # If provided, make sure output object is appropriate
-    if out is not None:
-        try:
-            data_parser(out, varname="out", writable=True, empty=True,
-                        dataclass="CrossSpectralData",
-                        dimord=st_dimord)
-        except Exception as exc:
-            raise exc
-        new_out = False
-    else:
-        out = CrossSpectralData(dimord=st_dimord)
-        new_out = True
+    out = CrossSpectralData(dimord=st_dimord)
 
     # now take the trial average from the single trial CR as input
     av_compRoutine.initialize(st_out, out._stackingDim, chan_per_worker=None)
-    av_compRoutine.pre_check() # make sure we got a trial_average
+    av_compRoutine.pre_check()   # make sure we got a trial_average
     av_compRoutine.compute(st_out, out, parallel=False, log_dict=log_dict)
 
-    # Either return newly created output object or simply quit
-    return out if new_out else None
+    # attach potential older cfg's from the input
```

Comment:
I meditated on this a little - I guess the only way to "really" automate this is to go back and let the CR's do the `cfg` updating. However, this would require a lot of namespace gymnastics to make things like `get_frontend_cfg` work. I'm not sure it's worth going through this kind of trouble to eliminate two lines (and in exchange probably adding 50 lines of `sys._getframet().code...` hacks). So agreed, let's stick with the boilerplate code.

Reply:
Also the CRs are not that helpful, as many of the frontend parameters already got digested (`process_taper` and so on..), such that replayability is not directly given and would also require some additional gymnastics in the frontend itself.